### PR TITLE
To display specific message when used /mention-each command

### LINKED ISSUE #240: When we use `/mention-each` command, the first line in the message should say: "Role mentioned: @xyz"
### Issue Description
When we use `/mention-each` command, the first line in the message should say:
"Role mentioned: @xyz"

### Expected Behavior
Athena bot should show "Role mentioned: [role here]" in the response.

### Current Behavior
Currently, `mention-each` does not show the role that was mentioned

### Screenshots
image

### Reproducibility

 - [ x ] This issue is reproducible
 - [ ] This issue is not reproducible

### Steps to Reproduce
1. Show the role that was mentioned in the query

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [ ] Medium
 - [ x ] Low

### Checklist

- [ x ] I have read and followed the project's code of conduct.
- [ x ] I have searched for similar issues before creating this one.
- [ x ] I have provided all the necessary information to understand and reproduce the issue.
- [ x ] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by RishiChaubey31**
Date: 20-10-24

Developer Name:  Rishi Chaubey

---

## Issue Ticket Number

- Closes #240
## Description

* A message is displayed  whenever user runs mention-each command,
* Put the feature behind feature flag
* written required tests for the feature
### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

Screenshot 2024-10-20 141052

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

Screenshot 2024-11-04 233859
Screenshot 2024-10-20 141919
Screenshot 2024-10-20 142045

</details>

## Additional Notes

recording of the whole working of the feature

https://github.com/user-attachments/assets/91bc3945-7263-4091-bf03-27de87837fa6

### CODE REVIEW

File: src/constants/commands.ts [hunk 1]
```diff
@@ -44,6 +44,12 @@ export const MENTION_EACH = {
       type: 3,
       require: false,
     },
+    {
+      name: "dev",
+      description: "want to tag them individually?",
```
Comment (vinit717) [inline src/constants/commands.ts:49]: Can you  change the description as this is feature flag

**Review by vinit717 — CHANGES_REQUESTED** [review]

File: src/constants/commands.ts (hunk 1 above)
Comment (RishiChaubey31) [reply src/constants/commands.ts]: done

File: src/constants/commands.ts [hunk 2]
```diff
@@ -50,6 +50,12 @@ export const MENTION_EACH = {
       type: 5,
       require: false,
     },
+    {
+      name: "showroles",
```
Comment (vinit717) [inline src/constants/commands.ts:54]: Can you confirm this with Ankush

File: src/controllers/mentionEachUser.ts [hunk 3]
```diff
@@ -36,13 +38,27 @@ export async function mentionEachUser(
...
+      responseMessage = `Sorry, no user found with <@&${roleId}> role.`;
+    } else if (usersWithMatchingRole.length === 1) {
+      // Mention the single user by ID
+      responseMessage = `The user with <@&${roleId}> role is ${payload.usersWithMatchingRole}.`;
+    } else {
+      // Mention multiple users by their IDs
+      responseMessage = `The users with <@&${roleId}> role are ${payload.usersWithMatchingRole}.`;
+    }
+    return discordTextResponse(responseMessage);
+  } else if (!dev || usersWithMatchingRole.length === 0) {
```
Comment (vinit717) [inline src/controllers/mentionEachUser.ts:53]: Too much nesting can you please fix this

**Review by vinit717 — COMMENTED** [review]
Please update the video after the optional argument name is finalize

File: tests/unit/handlers/mentionEachUser.test.ts [hunk 4]
```diff
@@ -42,7 +43,7 @@ describe("Test mention each function", () => {
       (res) => res.json()
     );
     expect(textMessage.data.content).toBe(
-      "Sorry no user found under this role."
+      `Sorry no user found with <@&${roleId}> role.`
```
Comment (vinit717) [inline tests/unit/handlers/mentionEachUser.test.ts:46]: This message is only shown when dev_titl is passed to true then how this test work?

File: tests/unit/handlers/mentionEachUser.test.ts [hunk 5]
```diff
@@ -126,4 +124,78 @@ describe("Test mention each function", () => {
...
+        name: "role",
+        type: 4,
+        value: roleId,
+      },
+      dev_title: {
+        name: "dev_title",
+        type: 4,
+        value: true,
+      },
+    };
```
Comment (vinit717) [inline tests/unit/handlers/mentionEachUser.test.ts:185]: Can you move this to fixture

File: tests/unit/handlers/mentionEachUser.test.ts [hunk 6]
```diff
@@ -126,4 +124,78 @@ describe("Test mention each function", () => {
...
+
+    const response = mentionEachUser(testData, env, ctx);
+    expect(response).toBeInstanceOf(Promise);
+
+    const textMessage: { data: { content: string } } = await response.then(
+      (res) => res.json()
+    );
+
+    // Since we can't mock getMembersInServer, both these messages are valid
```
Comment (vinit717) [inline tests/unit/handlers/mentionEachUser.test.ts:194]: Why can't we mock this?

Comment (RishiChaubey31) [reply tests/unit/handlers/mentionEachUser.test.ts:194]: we can mock this but I wanted to keep the changes as  similar as how the other  test were written

File: tests/unit/handlers/mentionEachUser.test.ts (hunk 5 above)
Comment (RishiChaubey31) [reply tests/unit/handlers/mentionEachUser.test.ts]: moved

File: src/controllers/mentionEachUser.ts [hunk 7]
```diff
@@ -36,13 +38,29 @@ export async function mentionEachUser(
...
+  }
+
   if (!dev || usersWithMatchingRole.length === 0) {
     const responseData = checkDisplayType({
       usersWithMatchingRole,
       msgToBeSent,
     });
     return discordTextResponse(responseData);
   } else {
+    // Regular dev flow to mention users
```
Comment (vinit717) [inline src/controllers/mentionEachUser.ts:63]: Do we need to add this line

File: tests/unit/handlers/mentionEachUser.test.ts
Comment (vinit717) [inline tests/unit/handlers/mentionEachUser.test.ts:1]: I think we don't need comments here make the test name a self explanatory

File: src/controllers/mentionEachUser.ts (hunk 7 above)
Comment (RishiChaubey31) [reply src/controllers/mentionEachUser.ts]: removed the comment

File: tests/unit/handlers/mentionEachUser.test.ts
Comment (RishiChaubey31) [reply tests/unit/handlers/mentionEachUser.test.ts:1]: removed the comment

**Comment by vinit717** [comment]
Can you add the test coverage not the test passing

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**